react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { currentIndex: 0 };
+    this.clickHandler = this.clickHandler.bind(this);
+  }
+
+  clickHandler(event) {
+    if (event.target.closest('div').classList.contains('arrow-left')) {
+      this.setState({ currentIndex: ((this.state.currentIndex + this.props.items.length) - 1) % this.props.items.length });
+      this.componentWillUnmount();
+    } else if (event.target.closest('div').classList.contains('arrow-right')) {
+      this.setState({ currentIndex: ((this.state.currentIndex + this.props.items.length) + 1) % this.props.items.length });
+      this.componentWillUnmount();
+    } else if (event.target.closest('ul').classList.contains('index')) {
+      this.setState({
+        currentIndex: (event.target.tagName === 'li')
+          ? parseInt(event.target.id, 10)
+          : parseInt(event.target.closest('li').id, 10)
+      });
+      this.componentWillUnmount();
+    }
+    this.componentDidMount();
+  }
+
+  componentDidMount() {
+    this.timer = setInterval(
+      () => this.setState({ currentIndex: ((this.state.currentIndex + this.props.items.length) + 1) % this.props.items.length }),
+      3000
+    );
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timer);
+  }
+
+  render() {
+    const keyList = this.props.items.map((element, index) => {
+      return (
+        <li key={element.index} id={index}>
+          {(this.state.currentIndex === index)
+            ? <i className="fas fa-circle"></i>
+            : <i className="far fa-circle"></i>}
+        </li>
```

Comment:
If you have a method for the dots you can put it directly on the li and even pass in the index directly:

```JSX
<li key={element.index} onClick={() => this.handleDotClick(index)}>
  {(this.state.currentIndex === index)
    ? <i className="fas fa-circle"></i>
    : <i className="far fa-circle"></i>}
</li>
```